patch-package
--- a/node_modules/detox/android/detox/build.gradle
+++ b/node_modules/detox/android/detox/build.gradle
@@ -2,7 +2,7 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion 25
-    buildToolsVersion '26.0.2'
+    buildToolsVersion '27.0.1'
     defaultConfig {
         minSdkVersion 18
         targetSdkVersion 25
@@ -12,10 +12,6 @@ android {
 
     productFlavors {
         flavorDimensions "minReactNative"
-        minReactNative44 {
-            dimension "minReactNative"
-
-        }
         minReactNative46 {
             dimension "minReactNative"
         }
@@ -57,9 +53,6 @@ android {
 }
 
 dependencies {
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp:3.4.1'
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp-ws:3.4.1'
-
     minReactNative46Implementation 'com.squareup.okhttp3:okhttp:3.6.0'
     minReactNative46Implementation 'com.squareup.okio:okio:1.13.0'
 
