patch-package
--- a/node_modules/detox/android/detox/build.gradle
+++ b/node_modules/detox/android/detox/build.gradle
@@ -19,10 +19,6 @@ android {
 
     productFlavors {
         flavorDimensions "minReactNative"
-        minReactNative44 {
-            dimension "minReactNative"
-
-        }
         minReactNative46 {
             dimension "minReactNative"
         }
@@ -64,9 +60,6 @@ android {
 }
 
 dependencies {
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp:3.4.1'
-    minReactNative44Implementation 'com.squareup.okhttp3:okhttp-ws:3.4.1'
-
     minReactNative46Implementation 'com.squareup.okhttp3:okhttp:3.6.0'
     minReactNative46Implementation 'com.squareup.okio:okio:1.13.0'
 
