patch-package
--- a/node_modules/react-native-firebase/android/build.gradle
+++ b/node_modules/react-native-firebase/android/build.gradle
@@ -16,13 +16,13 @@ apply plugin: 'com.android.library'
 
 android {
   compileSdkVersion 26
-  buildToolsVersion "25.0.3"
+  buildToolsVersion "27.0.1"
   defaultConfig {
     minSdkVersion 16
     targetSdkVersion 26
     versionCode 1
     versionName "1.0"
-    // multiDexEnabled true
+    multiDexEnabled true
   }
   buildTypes {
     release {
