patch-package
--- a/node_modules/react-native-navigation/android/app/build.gradle
+++ b/node_modules/react-native-navigation/android/app/build.gradle
@@ -3,7 +3,7 @@ apply from: '../prepare-robolectric.gradle'
 
 android {
     compileSdkVersion 25
-    buildToolsVersion "25.0.1"
+    buildToolsVersion "27.0.1"
 
     defaultConfig {
         minSdkVersion 16
new file mode 100644
--- /dev/null
+++ b/node_modules/react-native-navigation/react-native-navigation.podspec
@@ -0,0 +1,18 @@
+require "json"
+
+json = File.read(File.join(__dir__, "package.json"))
+package = JSON.parse(json).deep_symbolize_keys
+
+Pod::Spec.new do |s|
+  s.name = package[:name]
+  s.version = package[:version]
+  s.license = { type: "MIT" }
+  s.homepage = "https://github.com/wix/react-native-navigation"
+  s.authors = package[:author]
+  s.summary = package[:description]
+  s.source = { git: package[:repository][:url] }
+  s.source_files = "ios/**/*.{h,m}"
+  s.platform = :ios, "8.0"
+
+  s.dependency "React"
+end
